sympy/stats/stochastic_process_types.py

```diff
@@ -94,9 +95,9 @@ def _sym_sympify(arg):
         return _sympify(arg)
 
 def _matrix_checks(matrix):
-    if not isinstance(matrix, (Matrix, MatrixSymbol, ImmutableMatrix)):
+    if not isinstance(matrix, (MatrixBase, MatrixExpr)):
         raise TypeError("Transition probabilities either should "
-                            "be a Matrix or a MatrixSymbol.")
+                            "be a some type of Matrix.")
```

Comment:
The reason why only `Matrix`, `MatrixSymbol` and `ImmutableMatrix` were used was to be as specific as possible. We do not know in future if we introduce some new kind of matrix or any class which inherits `MatrixBase`. IMHO, it would be better to not touch those parts of code which aren't affecting current results or improving performance. This is not that big of a design change so let's not risk. It is better test but still why to change that part of code unless there is a very compelling reason to do that.

Reply:
You can add `FunctionMatrix` to the list but do not change the test. This may be a bit risky.

Reply:
@oscarbenjamin, this was your suggestion. Which one should I do? My knowledge is very limited on all the different matrix types.

Reply:
MatrixBase is the superclass of DenseMatrix and SparseMatrix which are in turn the superclasses for MutableDenseMatrix, ImmutableDenseMatrix, MutableSpaceMatrix and ImmutableSparseMatrix. Those are the four standard explicit matrix classes so anywhere we want to accept those we should test for MatrixBase.

Accepting only MatrixSymbol rather than MatExpr excludes other matrix expressions like `A**2` or `A*B` etc. I don't see why those couldn't be valid possibilities if MatrixSymbol is allowed.